### ISSUE TITLE
Feedback: it doesn't seem possible to limit search to the file name

### ISSUE THREAD

**Issue by Emstar** [Chaphasilor/odcrawler-frontend#7]
For example, say I want to look for books about "foo." Files would have "foo" in the filename, and the extension "epub". I search for "foo .epub". (maybe that isn't the right syntax, but it's my best guess.)

I would be most interested in seeing results like this:

- foo.epub
- Revenge of Foo.epub

The search will find those. But, the search results will return unwanted results like this:

- /files/**foo**/bar/Some Book.**epub**
- /some/path/**epub** and pdf/**foo**/Some Other Book.pdf
- /pub/How to Sell an **ePub** by Bob **Foo**.mobi
- and so on

I think as long as all search terms are anywhere in the path, it counts as a hit. That can make the list of results too long and hard to read. 

To find what you are looking for, it would be ideal to have an option to restrict the scope of the search to the file name. 

In fact, adding "search directory names" and "search file names" options in addition to the default "search all" would be super cool.

**Comment by Chaphasilor** [comment]
That is sadly not a frontend issue. I'm guessing @MCOfficer would have to manually specify the filename-part for each link and re-index all links in order for this to work.

Don't expect anything soon.

> I search for "foo .epub"

Btw, did you actually put a space between `foo` and `.epub`? If so, I think removing it might yield better results :)

**Comment by Emstar** [comment]
I tried all kinds of searches including "foo.epub." As far as I can tell, search works like this:

- Convert all punctuation in search terms to whitespace
- Look for all the search terms in the complete path of the indexed file
- Return any item that hits all terms, in any order, in any part of the path

So searching for "foo . epub" is the same as "foo.epub" and both are effectively a search for "foo epub".

That will hit files "foo.epub", "foo bar.epub" and "/some/path/foo/file.epub"

Totally makes sense that it is an index issue. If they don't give you the search parameters, nothing you can do at the time of query. 

You could build some kind of results list sort or filter feature, but that would clearly be a lot of work.

**Comment by MCOfficer** [comment]
this lines up with how i expect search engine to operate... split everything into words using a stemmer, then search those words. I don't want to change that because a lot of filenames use special characters instead of spaces for delimiting (think `some_name.txt`, `Timeline.2003.720p.BrRip.x264.YFY.mp4` etc)

I suppose the ideal solution would be google's: allow for quoted terms, so `foo.epub` will always search for `foo[ANY SPECIAL CHARACTER]epub`. Off the top of my head, i'm not sure if that's possible, though.

**Comment by Chaphasilor** [comment]
@MCOfficer if you have some spare time, if would be great if you check if it's possible, and if there are any changes required in the backend or my frontend request :)

**Comment by MCOfficer** [comment]
fair warning, there's a lot on my desk atm. Feel free to assign me on this.

**Comment by Chaphasilor** [comment]
take your time ;)

**Comment by Chaphasilor** [comment]
> I tried all kinds of searches including "foo.epub." As far as I can tell, search works like this:
> 
> * Convert all punctuation in search terms to whitespace
> * Look for all the search terms in the complete path of the indexed file
> * Return any item that hits all terms, in any order, in any part of the path
> 
> So searching for "foo . epub" is the same as "foo.epub" and both are effectively a search for "foo epub".
> 
> That will hit files "foo.epub", "foo bar.epub" and "/some/path/foo/file.epub"
> 
> Totally makes sense that it is an index issue. If they don't give you the search parameters, nothing you can do at the time of query.
> 
> You could build some kind of results list sort or filter feature, but that would clearly be a lot of work.

@Emstar @MCOfficer 
Okay, I have rather bad news:

I received [word from the MeiliSearch team](https://github.com/meilisearch/documentation/issues/531), it's not possible to search for an exact match and it's also not possible to prevent Meili from converting certain characters to a whitespace.  
And it gets even worse:
https://docs.meilisearch.com/guides/advanced_guides/datatypes.html#string

Basically, dots and certain other characters are considered as a *hard space*, which means they are treated as having a higher distance from each other, resulting in Meili almost completely disregarding that they should appear together.

That's a real bummer ://

**Comment by Chaphasilor** [comment]
On the upside, I'm currently designing the filename-only search :D

**Comment by Chaphasilor** [comment]
@Emstar 👀

![image](https://user-images.githubusercontent.com/18015147/96736428-e57ebd80-13bc-11eb-8ca6-35a5d9e41fcd.png)

**Comment by MCOfficer** [comment]
This is now live.

**Comment by Emstar** [comment]
This is great! Thanks for the hard work.
